src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,177 @@
+        LocalDate by = this.formatInputDate(deadline);
+        ui.handleInvalidInputDate(deadline, DateTimeFormatter.ofPattern("dd/MM/yyyy"), by);
+        return new Deadline(deadlineDescription, by);
+    }
+    
+    /**
+     * Creates an Event task from the input command, including parsing the start and end dates.
+     *
+     * @param input The user input string.
+     * @param ui The UI instance to handle user interactions.
+     * @return A new Event Task.
+     * @throws DukeException If any input date is invalid or the input format is incorrect.
+     */
+    public Task createEvent(String input, Ui ui) throws DukeException{
+        int indexOfFrom = input.indexOf("/from");
+        int indexOfTo = input.indexOf("/to");
+        String eventDescription = input.substring(6, indexOfFrom - 1);
+        String startString = input.substring(indexOfFrom + 6, indexOfTo - 1);
+        String endString = input.substring(indexOfTo + 4);
+        LocalDate start = this.formatInputDate(startString);
+        LocalDate end = this.formatInputDate(endString);
+        ui.handleInvalidInputDate(startString, DateTimeFormatter.ofPattern("dd/MM/yyyy"), start);
+        ui.handleInvalidInputDate(endString, DateTimeFormatter.ofPattern("dd/MM/yyyy"), end);
+        return new Event(eventDescription, start, end);
+    }
+    
+    /**
+     * Parses a string representation of a task from the storage file into a Task object.
+     *
+     * @param line The string representation of the task from the storage file.
+     * @return The corresponding Task object.
+     */
+    public Task parseStringToTask(String line) {
+        char typeOfTask = line.charAt(1);
+        char status = line.charAt(4);
+        Task t = null;
+        if (typeOfTask == 'T') {
+            String description = line.substring(7);
+            t = new ToDo(description);
+        } else if (typeOfTask == 'D') {
+            int indexOfTime = line.indexOf("(by: ");
+            String description = line.substring(7, indexOfTime - 1);
+            String byString = line.substring(indexOfTime + 5, line.indexOf(")"));
+            LocalDate by = parseFileDateToStorageDate(byString);
+            t = new Deadline(description, by);
+        } else if (typeOfTask == 'E') {
+            int indexOfStartTime = line.indexOf("(from");
+            int indexOfEndTime = line .indexOf("to");
+            String description = line.substring(7, indexOfStartTime - 1);
+            String startString = line.substring(indexOfStartTime + 6, indexOfEndTime - 1);
+            String endString = line.substring(indexOfEndTime + 3, line.indexOf(")"));
+            LocalDate start = parseFileDateToStorageDate(startString);
+            LocalDate end = parseFileDateToStorageDate(endString);
+            t = new Event(description, start, end);
+        }
+        boolean doneOrNot = (status == 'X');
+        t.setStatus(doneOrNot);
+        return t;
+    }
```

Comment:
Maybe can do the same as before, separate some of the lines inside this method with blank lines.